library/src/main/java/com/dslplatform/json/DslJson.java

```diff
@@ -1040,29 +1050,13 @@ public void registerBinder(final Type manifest, @Nullable final JsonReader.BindO
 	public JsonReader.ReadObject<?> tryFindReader(final Type manifest) {
 		JsonReader.ReadObject found = readers.get(manifest);
 		if (found != null) return found;
-		if (manifest instanceof Class && tryFindAndRegisterExternalConverter((Class)manifest)) {
-			found = readers.get(manifest);
-			if (found != null) return found;
-		}
 		for (ConverterFactory<JsonReader.ReadObject> rdr : readerFactories) {
 			found = rdr.tryCreate(manifest, this);
 			if (found != null) {
-				readers.put(manifest, found);
+				readers.putIfAbsent(manifest, found);
 				return found;
 			}
 		}
-		if (manifest instanceof ParameterizedType) {
-			ParameterizedType pt = (ParameterizedType) manifest;
```

Comment:
Not sure I follow how is this handled now

Reply:
I didn't find case where this is needed. I think each factory should handle ParameterizedType by their own.

Reply:
I guess we need a test for that.
The use case was that now `Optional<Something>` is not handled by default, but rather only after Optional is registered.
And it won't get registered for generic class, but rather only for container which will register new factories which know how to handle generic stuff in the future

Reply:
Yeah, I'll write one

Reply:
After last changes it behaves as expected, please rerun travis

Reply:
But you need to rebase? Travis will run without my test changes?

Reply:
I guess travis should test this pull request against latest commit in master branch

Reply:
great. tnx

Reply:
So I looked at it now and while it works "correctly" and loads up configuration only once, it doesn't work as I described above.
Meaning if you register your own factory for a type, it would get overridden with default one, which is not expected.
I'll make a test for that and change the behavior.

Reply:
You are right. I missed that case.